mirrorfolio/menteor
# (Bug fix) #21  Fix Server Actions Not Working in Production

### Description:
This pull request addresses issue #21 , where server actions were not functioning correctly in the production environment. The issue was diagnosed to be related to difficulties in accessing the backend in production, resulting in suppressed error logs and hindering root cause analysis. To resolve this, the backend setup was modified to ensure smooth operation in production by utilizing a Cloudflare worker to act as the backend, enabling seamless integration with Prisma and Neon.tech (PostgreSQL) as the stack. Additionally, the API endpoint was whitelabeled to accept requests from any source, enhancing flexibility and accessibility. This pull request not only fixes the bug causing issues with server actions in production but also provides a robust solution for future stability.

### Changes Made:
- Implemented modifications to the backend setup to facilitate server actions in the production environment.
- Utilized a Cloudflare worker to act as the backend, ensuring compatibility with Prisma and Neon.tech (PostgreSQL) stack.
- Whitelabeled the API endpoint to accept requests from any source, enhancing accessibility.
- Resolved the bug causing issues with server actions in production by configuring the Cloudflare worker to handle requests effectively.

### Testing:
- Tested the solution extensively in both development and production environments to ensure functionality and stability.
- Verified that server actions now work as expected in the production environment without encountering any errors or issues.

### How to Test:
1. Deploy the latest changes to the production environment.
2. Access the API endpoint: https://menteor.menteor-space.workers.dev/addToWaitingList
3. Send a POST request with the following form data:
   ```json
   {
     "formData":{
	     "name": "john brittas ",
	     "email": "john.brittas@gmail.com",
	     "phoneNumber": "0000000000",
	     "discoveryMethod": "other",
	     "exceptionalDiscovery": "reddit",
	     "getCallBack": true
	   }
   }
   ```
4. it returns the name of the user which is passed as a parameter to success screen 
5. Verify that the request is processed successfully, and the necessary actions are performed.

### Screenshots:



![image](https://github.com/mirrorfolio/menteor/assets/120118908/e384f369-6788-41f0-9e10-f176068a37e9)

------
<div>
    <a href="https://www.loom.com/share/e95c922b8a7e414c8946f065f1d7a686">
      <p>Menteor - Empowering Skill-Building Through Expert Mentorship - Google Chrome - 12 February 2024 - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/e95c922b8a7e414c8946f065f1d7a686">
      <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/e95c922b8a7e414c8946f065f1d7a686-with-play.gif">
    </a>
  </div>


### Checklist:
- [x] Implemented solution to fix the issue.
- [x] Tested changes thoroughly in both development and production environments.
- [x] Verified that server actions work as expected in production.
- [x] Documented the testing process and provided instructions for testing the changes.
